### PR TITLE
Adding test_apikeys.py and updating test_permission.py

### PR DESCRIPTION
**Pull request by koshy-thomas**
## Description

Added test_apikeys.py to cover tests for all the existing api methods for apikeys, also added some new methods which were missing in pyC8, corrected the url prefix to "/_api" for apikey related api.

Updated test_permission.py to cover tests for all the existing api methods for access levels(fabric, collection, streams, billing) and attributes, also added some new methods which were missing in pyC8

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

**Test Configuration**:

* C8 Version:

## Reviews

Please identify two developers to review this change

- [ ] @ricardo-macrometa 
- [x] @dlozina-macrometa 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added new functional tests that prove my fix is effective or that my feature works
- [x] Existing and new functional tests pass with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: c8/apikeys.py [hunk 1]
```diff
@@ -58,9 +63,28 @@ def response_handler(resp):
             else:
                 return resp.body
                 
-        return self._execute(request, response_handler)
+        return self._execute(request, response_handler, customPrefix="/_api")
```
Comment (dlozina-macrometa) [inline c8/apikeys.py:66]: It is not a huge error or even an error but is it possible to name customPrefix in a more pythonic(PEP8) way such as custom_prefix? If it is an issue we can perhaps do it later

File: c8/apikeys.py [hunk 2]
```diff
@@ -187,10 +211,36 @@ def response_handler(resp):
                 elif resp.body['error'] is True:
                     return False
                 
-        return self._execute(request, response_handler)
+        return self._execute(request, response_handler, customPrefix="/_api")
 
 #------------------------------------------------------------------------
```
Comment (dlozina-macrometa) [inline c8/apikeys.py:216]: Again not an issue, but can we remove these kinds of comments as we see them? No need for this brake, I have removed a couple of these as well. Thanks

File: c8/apikeys.py [hunk 3]
```diff
@@ -500,4 +550,95 @@ def response_handler(resp):
             else:
                 return resp.body
                 
-        return self._execute(request, response_handler)
+        return self._execute(request, response_handler, customPrefix="/_api")
+
+#-----------------------------------------------------------------------------
```
Comment (dlozina-macrometa) [inline c8/apikeys.py:555]: Same comment as above 😊

File: c8/exceptions.py [hunk 4]
```diff
@@ -675,7 +675,7 @@ class ClearStreamAccessLevel(C8ServerError):
 class ListStreams(C8ServerError):
     """Failed to fetch the streams for the specified key"""
 
-class BillingAcessLevel(C8ServerError):
+class BillingAccessLevel(C8ServerError):
```
Comment (dlozina-macrometa) [inline c8/exceptions.py:678]: Great catch! 😄

File: c8/tenant.py [hunk 5]
```diff
@@ -606,7 +607,7 @@ def get_database_access_level_user(self, username, databasename=""):
         def response_handler(resp):
             if resp.is_success:
                return resp.body['result']
-            raise DataBaseError(resp, request)
+            raise GetDataBaseAccessLevel(resp, request)
```
Comment (dlozina-macrometa) [inline c8/tenant.py:610]: I would add an else clause here. What do you think? We are raising an error if the response is NOK.

File: c8/tenant.py [hunk 6]
```diff
@@ -629,8 +630,8 @@ def remove_database_access_level_user(self, username, databasename=""):
 
         def response_handler(resp):
             if resp.is_success:
-               return resp.body
-            raise DataBaseError(resp, request)
+               return True
+            raise ClearDataBaseAccessLevel(resp, request)
```
Comment (dlozina-macrometa) [inline c8/tenant.py:634]: Same as the above 🔝

File: c8/tenant.py [hunk 7]
```diff
@@ -661,10 +662,37 @@ def set_database_access_level_user(self, username, databasename="", grant='ro'):
         def response_handler(resp):
             if resp.is_success:
                return resp.body
-            raise DataBaseError(resp, request)
+            raise SetDataBaseAccessLevel(resp, request)
```
Comment (dlozina-macrometa) [inline c8/tenant.py:665]: Same as the above 🔝

**Review by dlozina-macrometa — CHANGES_REQUESTED** [review]
Great job! I would love to hear your feedback on a couple of points that I mentioned.

File: c8/apikeys.py (hunk 1 above)
Comment (koshy-thomas) [reply c8/apikeys.py:66]: Sure I will do this in a separate PR tomorrow.

File: c8/tenant.py (hunk 5 above)
Comment (koshy-thomas) [reply c8/tenant.py:610]: Yes, we are raising an exception if the response is not a success (i.e. it contains a field called error: True). So I don't think an else is needed here

File: c8/tenant.py (hunk 6 above)
Comment (koshy-thomas) [reply c8/tenant.py:634]: Same as above

File: c8/tenant.py (hunk 7 above)
Comment (koshy-thomas) [reply c8/tenant.py:665]: Same as above

File: c8/apikeys.py (hunk 3 above)
Comment (koshy-thomas) [reply c8/apikeys.py:555]: Sure I will do this now

File: c8/apikeys.py (hunk 2 above)
Comment (koshy-thomas) [reply c8/apikeys.py:216]: Sure I will do this now

Comment (koshy-thomas) [reply c8/apikeys.py]: Done

File: c8/apikeys.py (hunk 3 above)
Comment (koshy-thomas) [reply c8/apikeys.py]: done

File: c8/apikeys.py (hunk 1 above)
Comment (dlozina-macrometa) [reply c8/apikeys.py:66]: 👍

File: c8/tenant.py (hunk 5 above)
Comment (dlozina-macrometa) [reply c8/tenant.py:610]: Great! Thanks

File: c8/tenant.py (hunk 6 above)
Comment (dlozina-macrometa) [reply c8/tenant.py:634]: 👍

**Comment by dlozina-macrometa** [comment]
Nice work! 💯

**Review by dlozina-macrometa — APPROVED** [review]
👍

**Review by ricardo-macrometa — APPROVED** [review]

**Pull request merged**